### PR TITLE
added suman

### PR DESCRIPTION
**Pull request by ORESoftware**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.

### CODE REVIEW

**Comment by ORESoftware** [comment]
Ok, so what does this mean:

```js
Error: Directory /home/travis/build/DefinitelyTyped/DefinitelyTyped/types/suman has unused file suman-tests.d.ts
    at /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/src/lib/definition-parser.ts:365:11
    at Generator.next (<anonymous>)
    at fulfilled (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/bin/lib/definition-parser.js:4:58)
    at <anonymous>
```

the test file is basically empty, because I don't really need tests for my typings.

**Comment by mhegazy** [comment]
@ORESoftware, please address the lint failures.

**Review by ORESoftware — COMMENTED** [review]
Looks good, I don't know why "declaration:false" is not good enough, I need to remove that entire line? Seems unnecessary?

File: types/suman/index.d.ts
Comment (andy-ms) [inline types/suman/index.d.ts]: It doesn't look like you changed anything about the template besides adding the package name?

File: types/suman/suman-tests.ts
Comment (andy-ms) [inline types/suman/suman-tests.ts]: It will need tests to ensure that code like https://github.com/sumanjs/suman#simple-example works.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by RyanCavanaugh** [comment]
@ORESoftware Please fix the failures indicated in the Travis CI log.

**Comment by ORESoftware** [comment]
will do thx

File: types/suman/index.d.ts
Comment (ORESoftware) [reply types/suman/index.d.ts]: I will do anything to get this to work, frankly I am lost, need help.

File: types/suman/suman-tests.ts
Comment (ORESoftware) [reply types/suman/suman-tests.ts]: what's wrong with console.log()? :) this is just a dummy PR, so that I know what to expect. An IMPORTANT QUESTION: Can I really not name any interface starting with an I? As in `interface IFooBar {}`... that is a really restrictive and unexpected limitation that forces me to refactor a lot of code :(

**Comment by ORESoftware** [comment]
@RyanCavanaugh @andy-ms 

I finally got the build to pass with TravisCI.

I am really loving the tslint tool, it's a great way to create consistent projects in the `@types` ecosystem.

I am having trouble getting tslint to work locally, if you can answer this, that would make my day better:
https://github.com/palantir/tslint/issues/3430

**Comment by RyanCavanaugh** [comment]
@andy-ms - Thanks for your review of this PR! Can you please look at the new code and update your review status if appropriate?

**Comment by RyanCavanaugh** [comment]
This PR has been open and unchanged 5 days without signoff or complaint. This will be merged by a maintainer soon if there are no objections.

**Comment by ORESoftware** [comment]
SGTM thanks

File: types/suman/index.d.ts
Comment (andy-ms) [reply types/suman/index.d.ts]: It looks like suman provides its own typings in `lib/index.d.ts`. You shouldn't need to add these to DefinitelyTyped when they are already bundled with the package.

**Comment by andy-ms** [comment]
Closing as `suman` provides its own types.

**Comment by ORESoftware** [comment]
@andy-ms, suman doesn't provide all its own types.  This project, suman-types provides the types that are not included in suman itself -https://github.com/sumanjs/suman-types

and I was basically wanting to publish a git subtree of suman-types to `@types/suman`.

Why is this better? Because then people don't have to use anything more than the standard `typeRoots`.

**Comment by andy-ms** [comment]
Why not make a pull request to `suman` adding the missing types?

**Comment by ORESoftware** [comment]
@andy-ms the sumanjs ecosystem is made up of 10+ projects. The dependency tree of that ecosystem is complex. The reason why I don't want to include all the types in suman itself, is that in some cases I do not want to `npm link` the suman project to certain projects locally, instead I wish to `npm link`the  suman-types project because it's lighter-weight - just the typings.... So `suman-types` serves as a lighter-weight shared dependency of just the types for the whole sumanjs ecosystem.

**Comment by ORESoftware** [comment]
But it seems like I should bypass `@types`....I can include `suman-types` in the `suman` project, and the export the types from there. This is all very confusing but one thing is becoming more certain...making PRs and waiting for PRs to `@types` is a little bit stressful and the loop is slow.

**Comment by andy-ms** [comment]
Yes, DefinitelyTyped is just for packages that didn't come bundled with their own types -- if this is your own project you should include types yourself.

**Comment by ORESoftware** [comment]
yeah np, it's a little confusing, but like I said I can include `suman-types` in my` suman` project and then export the types from `suman-types` from `suman`. Thanks anyway.

**Pull request closed without merging**
